app/models/event.rb

```diff
@@ -18,13 +18,19 @@ def self.find_by_path(path, base: Rails.root.join("_events"), try_index: false)
     super path, base: base, try_index: try_index
   end
 
+  attr_reader :filename
+  has_field :title
+  has_field :speakers, :audience, default: []
+  has_field :accommodations, default: {}
+
   def initialize(path, base: nil)
+    @filename = path.basename(".md")
     loader = FrontMatterParser::Loader::Yaml.new(allowlist_classes: [Time])
     @parsed_file = FrontMatterParser::Parser.parse_file(path, loader: loader)
   end
 
-  def title
-    parsed_file["title"]
+  def to_param
```

Comment:
The `to_param` method is what enables the view to figure out how to build the identifier part of the URL with `event_path(event)`